### PR TITLE
move device-specific teardown logic from training loop to accelerator

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Follow up to #5743 

on_train_end device-specific teardown should be handled by accelerator. 
Makes training loop device agnostic

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/accelerators/gpu.py [hunk 1]
```diff
@@ -27,6 +27,7 @@ def on_train_start(self):
 
     def on_train_end(self):
         # clean up memory
+        self.model.cpu()
         with torch.cuda.device(self.root_device):
             torch.cuda.empty_cache()
```
Comment (carmocca) [inline pytorch_lightning/accelerators/gpu.py:32]: question: the `with` block is to empty only the cache of the root device? I thought `empty_cache` emptied all

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/accelerators/gpu.py (hunk 1 above)
Comment (awaelchli) [reply pytorch_lightning/accelerators/gpu.py:32]: did a quick search, more confused than before, but found that soumith suggests it here:
https://github.com/pytorch/pytorch/issues/25752#issuecomment-528866347

**Review by Borda — APPROVED** [review]

File: pytorch_lightning/accelerators/gpu.py [hunk 2]
```diff
@@ -27,6 +27,7 @@ def on_train_start(self):
 
     def on_train_end(self):
         # clean up memory
+        self.model.cpu()
```
Comment (tchaton) [inline pytorch_lightning/accelerators/gpu.py:30]: Should we do this for TPU too ?

File: pytorch_lightning/trainer/training_loop.py [hunk 3]
```diff
@@ -148,13 +148,7 @@ def on_train_end(self):
             self.trainer.profiler.describe()
 
         # give accelerators a chance to finish
-        self.trainer.accelerator_backend.on_train_end()
-
-        # clear mem
-        if self.trainer._device_type == DeviceType.GPU:
```
Comment (tchaton) [inline pytorch_lightning/trainer/training_loop.py:154]: Nice cleaning !

**Review by tchaton — APPROVED** [review]
LGTM !
